graphics/pixelbuffer.cpp

```diff
@@ -88,14 +88,34 @@ void PixelBuffer::clear(int length) {
 
 void PixelBuffer::setPixelAt(int pixel, uint32 value) {
 #if defined(SCUMM_BIG_ENDIAN)
-	byte *buffer = _buffer + pixel * _format.bytesPerPixel;
-	for (int i = 0; i < _format.bytesPerPixel; ++i) {
-		buffer[i] = value >> ((_format.bytesPerPixel - i - 1) * 8) & 0xFF;
+	switch (_format.bytesPerPixel) {
+	case 2:
+		((uint16 *) _buffer)[pixel] = (uint16) ((value << 8) | (value >> 8));
+		break;
+	case 4:
+		((uint32 *) _buffer)[pixel] = (uint32) (((value & 0xFF00FF00) >> 8) | ((value & 0x00FF00FF) << 8));
```

Comment:
Is this line supposed to be a 32 bits endian swap?
If so, I would suggest using the SWAP_BYTES_32 macro, which will use a compiler builtin in most cases.